### PR TITLE
Fix docker.run

### LINKED ISSUE #17088: state.dockerio.run: docked_onlyif and docked_unless do not work
```docked_onlyif``` and ```docked_unless``` is usable to make the state idempotent.

With the following state:
```
thecommand:
  docker.run:
    - cid: container
```
```thecommand``` will always be executed in the docker container.

With the following state:
```
thecommand:
  docker.run:
    - cid: container
    - docked_unless: /bin/true
```
```thecommand``` will not be executed because ```/bin/true``` indicates the system is already converged in the desired state.

However, ```docked_onlyif``` disables the function ```state.dockerio.run``` entirely and ```docked_unless``` have totally no effects in the current implement.

This malfunction is caused by the bug in the function ```modules.dockerio.retcode```.
```modules.dockerio.retcode``` should return the return code as ```int```, but it returns an instance of ```dict```.
I created the minimum case to prove ```modules.dockerio.retcode``` returns ```dict```:
https://github.com/umireon/salt-docker-retcode-is-broken-vagrant
and the log:
https://gist.github.com/umireon/baa589bf1d74ce3b9f7e

### PR DESCRIPTION
**Pull request by Arabus**
This should fix #17088 - not sure about the last commit whether cmd.run really is the way to go to run tests on docker hosts.

### CODE REVIEW

**Comment by jfindlay** [comment]
@Arabus, thanks for the fix.  There is some [lint](https://jenkins.saltstack.com/job/salt-pr-lint-n/9062/violations/file/salt/modules/dockerio.py/).  Also, starting with 2015.8.0, dockerio is deprecated in favor of dockerng.

**Comment by Arabus** [comment]
@jfindlay that lint was to be expected, the module and state code are in a bad shape - I will spend some time this afternoon fixing them up and then have a look at the dockerng code, whether it suits my needs :-)

**Comment by Arabus** [comment]
There, @jfindlay that should do it :-) I'll have a look at the dockerng module now.  Does your comment mean the dockerio module will be replaced or will dockerng be integrated as an alternative?

**Comment by jfindlay** [comment]
@Arabus, the [dockerio documentation](https://docs.saltstack.com/en/2015.8/ref/modules/all/salt.modules.dockerio.html) for 2015.8 has a note about dockerio being deprecated.  For now both sets of modules will exist simultaneously, but [beginning with 2015.8](https://docs.saltstack.com/en/2015.8/topics/releases/2015.8.0.html#state-and-execution-module-improvements), we will be encouraging people to migrate to dockerng.

File: salt/states/dockerio.py [hunk 1]
```diff
@@ -471,7 +471,7 @@ def loaded(name, source=None, source_hash='', force=False):
         remove_info = remove_image(name)
         if not remove_info['status']:
             return _invalid(name=name,
-                    comment='Image could not be removed: {0}'.format(name))
+                            comment='Image could not be removed: {1}'.format(name))
```
Comment (jfindlay) [inline salt/states/dockerio.py]: This needs to be `{0}`.

Comment (Arabus) [inline salt/states/dockerio.py]: weird I cannot remember actually changing the value of that - I did change the indentation for linting though. Ill fix that

**Comment by Arabus** [comment]
harrumpf not one of my best works :-/  this should fix it

**Pull request merged**